### PR TITLE
Add accessablity with hoc tests

### PR DESCRIPTION
**Pull request by meltedspork**

### CODE REVIEW

**Comment by meltedspork** [comment]
I think we are getting closer

**Comment by kratsg** [comment]
Why is the HOC wrapped in a function? Is the wrapped function needed?

**Comment by meltedspork** [comment]
> Why is the HOC wrapped in a function? Is the wrapped function needed?

LOL. i was so tired last night that i forgot to refactor this. Ill fix that in moment

**Comment by meltedspork** [comment]
@kratsg PTAL - now only thing left is testing the ehance-accessbliity tests!

**Comment by meltedspork** [comment]
Actually - i think we are going wrong way around about this with the test for HOC. Give me about a day with that last part

File: src/containers/hoc/enhance-accessibility.jsx [hunk 1]
```diff
@@ -2,13 +2,14 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-import { getDisplayName } from '../utils/getDisplayName';
-import { a11yActions } from '../redux/modules/a11y';
+import getDisplayName from '../../utils/getDisplayName';
+import { a11yActions } from '../../redux/modules/a11y';
 
-const AccessibleHOC = (WrappedComponent) => {
+function withAccessible(WrappedComponent) {
```
Comment (kratsg) [inline src/containers/hoc/enhance-accessibility.jsx]: change to `const withAccessible = (wrappedComponent) => {`

File: src/containers/hoc/enhance-accessibility.jsx [hunk 2]
```diff
@@ -42,6 +43,8 @@ const AccessibleHOC = (WrappedComponent) => {
     setA11yNavigatedMessage: PropTypes.func.isRequired,
   };
 
+  ScreenReaderAccessibleComponent.WrappedComponent = WrappedComponent;
```
Comment (kratsg) [inline src/containers/hoc/enhance-accessibility.jsx]: Shouldn't this be part of the constructor? Different instances would share the same class, so we want to set the `WrappedComponent` on the instance, not on the class.

Comment (meltedspork) [reply src/containers/hoc/enhance-accessibility.jsx]: TIL: dont need do constructor (this doesn't work anyways in this case but can do)
`static WrappedComponent = WrappedComponent;` inside the class

File: src/containers/hoc/enhance-accessibility.jsx [hunk 3]
```diff
@@ -5,13 +5,16 @@ import PropTypes from 'prop-types';
 import getDisplayName from '../../utils/getDisplayName';
 import { a11yActions } from '../../redux/modules/a11y';
 
-const withAccessible = (WrappedComponent) => {
-  class ScreenReaderAccessibleComponent extends React.Component {
+function makeAccessibleRouteComponent(WrappedComponent) {
```
Comment (kratsg) [inline src/containers/hoc/enhance-accessibility.jsx]: arrow style here or?

Comment (meltedspork) [reply src/containers/hoc/enhance-accessibility.jsx]: 👍👍👍 done

**Pull request merged**